src/plugins/intel_gpu/src/plugin/transformations_pipeline.cpp

```diff
@@ -407,6 +408,9 @@ void TransformationsPipeline::apply(std::shared_ptr<ov::Model> func) {
             }
         }
 
+        if (infer_precision == ov::element::f16) {
```

Comment:
I think `infer_precision != ov::element::f32` is correct condition here because it can be `dynamic`.